src/test/scala/ai/privado/exporter/SourceExporterTest.scala

```diff
@@ -53,24 +48,83 @@ class SourceExporterTest extends JavaFrontendTestSuite {
         .nonEmpty shouldBe true
     }
 
-    "not export derived source under processing" in {
-      val dataflowMap =
-        Dataflow(cpg, new StatsRecorder()).dataflow(privadoInput, ruleCache, dataflowCache, auditCache, appCache)
-      val sourceExporter =
-        SourceExporter(
-          cpg,
-          ruleCache,
-          privadoInput,
-          appCache = appCache,
-          dataFlowCache = dataflowCache,
-          dataflows = dataflowMap
-        )
-      !sourceExporter.getProcessing.flatMap(_.occurrences).map(_.sample).exists(_.equals("user")) shouldBe true
-      sourceExporter.getProcessing.map(_.sourceId).headOption.get shouldBe "Data.Sensitive.FirstName"
-      sourceExporter.getProcessing
-        .flatMap(_.occurrences)
-        .map(_.sample)
-        .exists(_.equals("java.lang.String firstName")) shouldBe true
+    "export derived source under processing" in {
+      val outputJson     = cpg.getPrivadoJson()
+      val processingList = getProcessings(outputJson)
+
+      // should be present inside processing section because "user" is the first node of dataflow
+      processingList.flatMap(_.occurrences).map(_.sample).exists(_.equals("user")) shouldBe true
+    }
+  }
+
+  "when derived node is start node" should {
+    val privadoInput = PrivadoInput()
+    val cpg = code(
+      """
+        |class User {
+        |   public String firstName;
+        |}
+        |
+        |class Main {
+        |  public void printValue() {
+        |  User user = new User();
+        |  System.out.println(user);
+        |  }
+        |}
+        |""".stripMargin,
+      "index.java"
+    )
+      .withRuleCache(ruleCache)
+      .withPrivadoInput(privadoInput)
+
+    "tag a derived source" in {
+      val derivedSource = cpg.identifier("user").lineNumber(8).l
+      derivedSource.tag
+        .nameExact(Constants.catLevelOne)
+        .valueExact(CatLevelOne.DERIVED_SOURCES.name)
+        .nonEmpty shouldBe true
+    }
+
+    "Processing section should have firstNode of every dataflow" in {
+      val outputJson     = cpg.getPrivadoJson()
+      val processingList = getProcessings(outputJson)
+
+      processingList.headOption.get.occurrences.size shouldBe 2
+      processingList.map(_.sourceId).exists(_.equals("Data.Sensitive.FirstName")) shouldBe true
+
+      processingList.flatMap(_.occurrences).map(_.sample).exists(_.equals("user")) shouldBe true
```

Comment:
`Processing section should have firstNode of every dataflow` Also add the dataflow validation here. So that we can confirm if the first node in dataflow is actually coming in processing

Reply:
yes done.